Compatability with Python 3.12

I had a user trying to build with python 3.12. Currently fails with
ndicapimodule.cxx:29:41: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
   29 |   #define PY_INT_OBJECT_OB_IVAL(ob) ob->ob_digit[0]

https://github.com/SciKit-Surgery/ndicapi/actions/runs/7991400661/job/21822469379#step:9:379

--- a/ndicapimodule.cxx
+++ b/ndicapimodule.cxx
@@ -26,7 +26,11 @@
   #define PyString_AsString PyUnicode_AsUTF8
   #define PyIntObject PyLongObject
   //#define PY_INT_OBJECT_OB_IVAL(ob) PyLong_AsLong((PyObject*)(ob))
-  #define PY_INT_OBJECT_OB_IVAL(ob) ob->ob_digit[0]
+  #if PY_MINOR_VERSION < 12
+    #define PY_INT_OBJECT_OB_IVAL(ob) ob->ob_digit[0]
+  #else
+    #define PY_INT_OBJECT_OB_IVAL(ob) ob->long_value.ob_digit[0]
+  #endif
   #define cmpfunc PyAsyncMethods*
 #else
   #define MOD_ERROR_VAL

